### PR TITLE
feat(HACBS-154): add release metrics

### PR DESCRIPTION
**Pull request by davidmogar**
This commit adds metrics to the release resource so we can support the Release Service SLIs. In addition, adds changes to allow the Prometheus instance in openshift-monitoring to scrap our ServiceMonitor.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,3 +51,16 @@ using the following command:
...
 ```
+
+## Metrics
+
+Apart from the [metrics provided by controller-runtime](https://book.kubebuilder.io/reference/metrics-reference.html)
+by default, this operator exports the following custom metrics:
+
+| Metric name                      | Type      | Description                                                                |
+|----------------------------------|-----------|----------------------------------------------------------------------------|
+| release_attempt_concurrent_total | Gauge     | Total number of concurrent releases.                                       |
```
[the remark this answers is not in the document]
Comment (davidmogar) [reply README.md:62]: Gauge is one of the metric types supported by Prometheus. You can see them all [here](https://prometheus.io/docs/concepts/metric_types/).

Regarding the concurrent metric, the way it works is by increasing and decreasing once a new release start and finishes respectively.

File: README.md [hunk 2]
```diff
@@ -51,3 +51,16 @@ using the following command:
...
+| release_attempt_duration_seconds | Histogram | Total duration of the release from the moment the PipelineRun was created. |
```
Comment (johnbieren) [inline README.md:63]: As talked about, the description should be updated to be more clear

**Comment by davidmogar** [comment]
> `feat(HACBS-154): enable openshift-monitoring scrapping` I think you mean `feat(HACBS-154): enable openshift-monitoring scraping` (not sure how to comment on the commit title)

Totally, good catch. Modified.

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (johnbieren) [reply README.md:62]: As talked about on the call, let's pick a new name. I don't have one in mind yet

File: README.md [hunk 3]
```diff
@@ -51,3 +51,16 @@ using the following command:
...
+| release_attempt_invalid_total    | Counter   | Number of invalid releases.                                                |
+| release_attempt_running_seconds  | Histogram | Time for a release to change its state to running.                         |
```
Comment (johnbieren) [inline README.md:65]: The description should be updated to be more clear

Comment (davidmogar) [reply README.md]: Changed to `Time for releases to change their state to running`. Not sure it's even better. What do you think?

File: README.md (hunk 2 above)
Comment (davidmogar) [reply README.md]: Changed to `Releases duration from the moment the release PipelineRun was created`. Not sure it's better though. Suggestions accepted.

**Comment by davidmogar** [comment]
I cannot add a reply to your Gauge metric comment but I can edit the comment itself. Curious... Any way, changed to `release_attempt_concurrent_requests`. Let me know if this name is better.

**Comment by johnbieren** [comment]
> I cannot add a reply to your Gauge metric comment but I can edit the comment itself. Curious... Any way, changed to `release_attempt_concurrent_requests`. Let me know if this name is better.

I resolved it. I am good with the new name

File: README.md (hunk 3 above)
Comment (johnbieren) [reply README.md]: I like it better, but maybe like (and you'll probably hate me for suggesting the word total now after complaining about it before) `Total time for all releases to change their state to running from created`

File: README.md (hunk 2 above)
Comment (johnbieren) [reply README.md]: `Total time for all releases to change their state to finished, starting from the moment their release PipelineRun was created` ? That could probably be better English... :man_shrugging: 

And to make sure I remember this correctly, it is state running to state completed, so the processing/validation time in the operator is not included?

**Review by johnbieren — APPROVED** [review]

**Pull request merged**